src/OneDrive.Sdk.Authentication.Common/MsaAuthenticationProvider.cs

```diff
@@ -295,6 +295,36 @@ protected void DeleteUserCredentialsFromCache(AccountSession accountSession)
         }
 
         /// <summary>
+        /// Retrieves the authentication token. Retrieves the most recently
+        /// used credentials if available, without showing the sign in UI if credentials are unavailable.
+        /// </summary>
+        /// <param name="userName">The login name of the user, if known.</param>
+        /// <returns>The authentication token.</returns>
+        public async Task RestoreMostRecentFromCacheAsync(string userName = null)
+        {
+            using (var httpProvider = new HttpProvider())
+            {
+                await this.RestoreMostRecentFromCacheAsync(httpProvider, userName).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the authentication token. Retrieves the most recently
+        /// used credentials if available, without showing the sign in UI if credentials are unavailable.
+        /// </summary>
+        /// <param name="httpProvider">HttpProvider for any web requests needed for authentication</param>
+        /// <param name="userName">The login name of the user, if known.</param>
+        /// <returns>The authentication token.</returns>
+        public async Task RestoreMostRecentFromCacheAsync(IHttpProvider httpProvider, string userName = null)
```

Comment:
If this is going to be public, I think it ought to return a `bool` indicating whether something was retrieved from the cache or not. Return `true` iff `authResult !=null`. That way the client can more concisely decide how to act on the result (instead of turning right around and calling `IsAuthenticated`).